### PR TITLE
Update `PointCloud` to conform to requirements of `ReversibleContainer`

### LINKED ISSUE #2814: Add const interators in PointCloud to cbegin cend
```
apps/in_hand_scanner/src/icp.cpp:407:33: error: no member named 'cbegin' in 'pcl::PointCloud<pcl::PointXYZRGBNormal>'; did you mean 'begin'?
  for (auto it_in = cloud_data->cbegin (); it_in!=cloud_data->cend (); ++it_in)
```

And pointcloud.h:
```
      inline iterator begin () { return (points.begin ()); }
      inline iterator end ()   { return (points.end ()); }
      inline const_iterator begin () const { return (points.begin ()); }
      inline const_iterator end () const  { return (points.end ()); }
```
`cloud_data` is const, so he should use correct version.

_Originally posted by @SunBlack in https://github.com/_render_node/MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDI1MjIwMDE1OA==/comments/review_comment_

### LINKED ISSUE #3966: [common] Add member functions cbegin() and cend() for accessing const_iterator in PointCloud
Add member functions `cbegin()` and `cend()` to return `const_iterator` for pointclouds.
Currently to use `const_iterator` it has to be explicitly declared and you cannot use `auto`.

**Expected behavior**

```cpp
for (auto it = cloud.cbegin(); it != cloud.cend(); ++it ) { .. }
```

**Current Behavior**

```cpp
for (PointCloud<PointXYZ>::const_iterator it = cloud.cbegin(); it != cloud.cend(); ++it ) { .. }
```
or 

```cpp
for (auto it = cloud.points.cbegin(); it != cloud.points.cend(); ++it ) { .. }
```

### PR DESCRIPTION
**Pull request by shrijitsingh99**
Resolves #3966 
Closes #2814

### CODE REVIEW

File: common/include/pcl/point_cloud.h [hunk 1]
```diff
@@ -439,10 +439,16 @@ namespace pcl
...
       using const_iterator = typename VectorType::const_iterator;
+      using reverse_iterator = typename VectorType::reverse_iterator;
+      using const_reverse_iterator = typename VectorType::const_reverse_iterator;
       inline iterator begin () { return (points.begin ()); }
       inline iterator end ()   { return (points.end ()); }
-      inline const_iterator begin () const { return (points.begin ()); }
-      inline const_iterator end () const  { return (points.end ()); }
+      inline const_iterator begin () const { return (points.cbegin ()); }
+      inline const_iterator end () const  { return (points.cend ()); }
+      inline reverse_iterator rbegin () { return (points.rbegin ()); }
```
Comment (kunaltyagi) [inline common/include/pcl/point_cloud.h:448]: Needs const overload aka `rbegin() const`

File: common/include/pcl/point_cloud.h [hunk 2]
```diff
@@ -439,10 +439,16 @@ namespace pcl
       // iterators
       using iterator = typename VectorType::iterator;
       using const_iterator = typename VectorType::const_iterator;
+      using reverse_iterator = typename VectorType::reverse_iterator;
+      using const_reverse_iterator = typename VectorType::const_reverse_iterator;
       inline iterator begin () { return (points.begin ()); }
       inline iterator end ()   { return (points.end ()); }
-      inline const_iterator begin () const { return (points.begin ()); }
-      inline const_iterator end () const  { return (points.end ()); }
+      inline const_iterator begin () const { return (points.cbegin ()); }
```
Comment (kunaltyagi) [inline common/include/pcl/point_cloud.h:446]: Mark these as `noexcept`

File: common/include/pcl/point_cloud.h [hunk 3]
```diff
@@ -439,10 +439,16 @@ namespace pcl
...
+      inline const_iterator end () const  { return (points.cend ()); }
```
Comment (kunaltyagi) [inline common/include/pcl/point_cloud.h:447]: Add `cbegin` and `cend`

File: common/include/pcl/point_cloud.h [hunk 4]
```diff
@@ -439,10 +439,20 @@ namespace pcl
...
       using const_iterator = typename VectorType::const_iterator;
-      inline iterator begin () { return (points.begin ()); }
-      inline iterator end ()   { return (points.end ()); }
-      inline const_iterator begin () const { return (points.begin ()); }
-      inline const_iterator end () const  { return (points.end ()); }
+      using reverse_iterator = typename VectorType::reverse_iterator;
+      using const_reverse_iterator = typename VectorType::const_reverse_iterator;
+      inline iterator begin () noexcept { return (points.begin ()); }
+      inline iterator end () noexcept { return (points.end ()); }
+      inline iterator begin () const noexcept { return (points.begin ()); }
```
Comment (kunaltyagi) [inline common/include/pcl/point_cloud.h:446]: const overloads will return const_iterators. See [vector](https://en.cppreference.com/w/cpp/container/vector/begin) for reference

**Comment by kunaltyagi** [comment]
Please check the CI

**Comment by shrijitsingh99** [comment]
Retriggering CI since the logs got deleted.

**Comment by shrijitsingh99** [comment]
CI seems to be green now. 
The macOS might have been failing due to the SDK error in the Mac CI which was present for a brief period.

**Pull request merged**